python-sdk/nuscenes/lidarseg/lidarseg_utils.py

```diff
@@ -0,0 +1,228 @@
+    to class color, produce a legend which shows the color and the corresponding class name.
+    :param labels_to_include_in_legend: Labels to show in the legend.
+    :param idx2name: The mapping from class index to class name.
+    :param name2color: The mapping from class name to class color.
+    :param loc: The location of the legend.
+    :param ncol: The number of columns that the legend has.
+    :param bbox_to_anchor: A 2-tuple (x, y) which places the top-left corner of the legend specified by loc
+                           at x, y. The origin is at the bottom-left corner and x and y are normalized between
+                           0 and 1 (i.e. x > 1 and / or y > 1 will place the legend outside the plot.
+    """
+
+    recs = []
+    classes_final = []
+    classes = [name for idx, name in sorted(idx2name.items())]
+
+    for i in range(len(classes)):
+        if labels_to_include_in_legend is None or i in labels_to_include_in_legend:
+            name = classes[i]
+            recs.append(mpatches.Rectangle((0, 0), 1, 1, fc=np.array(name2color[name]) / 255))
+
+            # Truncate class names to only first 25 chars so that legend is not excessively long.
+            classes_final.append(classes[i][:25])
+
+    plt.legend(recs, classes_final, loc=loc, ncol=ncol, bbox_to_anchor=bbox_to_anchor)
+
+
+def paint_points_label(lidarseg_labels_filename: str, filter_lidarseg_labels: List[int],
+                       name2idx: Dict[str, int], colormap: Dict[str, Tuple[int, int, int]]) -> np.ndarray:
+    """
+    Paint each label in a pointcloud with the corresponding RGB value, and set the opacity of the labels to
+    be shown to 1 (the opacity of the rest will be set to 0); e.g.:
+        [30, 5, 12, 34, ...] ------> [[R30, G30, B30, 0], [R5, G5, B5, 1], [R34, G34, B34, 1], ...]
+    :param lidarseg_labels_filename: Path to the .bin file containing the labels.
+    :param filter_lidarseg_labels: The labels for which to set opacity to zero; this is to hide those points
+                                   thereby preventing them from being displayed.
+    :param name2idx: A dictionary containing the mapping from class names to class indices.
+    :param colormap: A dictionary containing the mapping from class names to RGB values.
+    :return: A numpy array which has length equal to the number of points in the pointcloud, and each value is
+             a RGBA array.
+    """
+
+    # Load labels from .bin file.
+    points_label = np.fromfile(lidarseg_labels_filename, dtype=np.uint8)  # [num_points]
+
+    # Given a colormap (class name -> RGB color) and a mapping from class name to class index,
+    # get an array of RGB values where each color sits at the index in the array corresponding
+    # to the class index.
+    colors = colormap_to_colors(colormap, name2idx)  # Shape: [num_class, 3]
+
+    if filter_lidarseg_labels is not None:
```

Comment:
assertion for filter_lidarseg_labels not to be None since we do not provide a default value for it?

Reply:
@whyekit-aptiv ?

Reply:
If an user does `filter_lidarseg_labels=None`, all classes will be shown, so I don't think we need an assertion here